### ISSUE TITLE
Module not found: Error: Cannot resolve module 'fs'

### ISSUE THREAD

**Issue by rowino** [BigstickCarpet/json-schema-ref-parser#27]
I am using the babel [generator boilerplate](https://github.com/babel/generator-babel-boilerplate) for a library I am building. I added this library and imported it into my project. Now when I run gulp build I get the following error. Any assistance would be highly appreciated.

```
stream.js:74
      throw er; // Unhandled stream error in pipe.
      ^
Error: ./~/json-schema-ref-parser/lib/resolvers/file.js
Module not found: Error: Cannot resolve module 'fs' in /home/raphael/work/projects/packages/json-schema-form-generator/node_modules/json-schema-ref-parser/lib/resolvers
resolve module fs in /home/raphael/work/projects/packages/json-schema-form-generator/node_modules/json-schema-ref-parser/lib/resolvers
  looking for modules in /home/raphael/work/projects/packages/json-schema-form-generator/node_modules
    /home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /home/raphael/work/projects/packages/json-schema-form-generator/node_modules
      resolve file
        /home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs doesn't exist
        /home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs.webpack.js doesn't exist
        /home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs.web.js doesn't exist
        /home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs.js doesn't exist
        /home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs.json doesn't exist
[/home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs]
[/home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs]
[/home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs.webpack.js]
[/home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs.web.js]
[/home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs.js]
[/home/raphael/work/projects/packages/json-schema-form-generator/node_modules/fs.json]
 @ ./~/json-schema-ref-parser/lib/resolvers/file.js 2:14-27
```

**Comment by BigstickCarpet** [comment]
It looks like you're using webpack, and something in your config is causing it to load the Node.js version of JSON Schema Ref Parser rather than the browser version.

**Comment by mari0juric** [comment]
I had the same problem. Solved it by adding the alias in webpack so it knows where to find the correct asset.

`resolve : {
        alias: {
            'json-schema-parser': __dirname + '/node_modules/json-schema-ref-parser/dist/ref-parser.js'
        }
    }`

and you use it as you would normally but use the alias key

`var parser = require('json-schema-parser')`

**Comment by BigstickCarpet** [comment]
Closing this issue, since it's a WebPack config issue.  Thanks to @mari0juric for posting a solution, which will help others who are facing this problem. 👍

**Comment by iamakulov** [comment]
FYI: the latest package on npm doesn’t have the `browser` field for some reason: https://unpkg.com/json-schema-ref-parser@3.1.2/package.json. This breaks my webpack build.

Probably bumping the version and re-releasing the package could fix the problem? Or is the field removed during the publish routine (`npm run release`)?

**Comment by iamakulov** [comment]
Ah, yeah, it’s not published because the latest release was before #30 was merged.

Is there a chance that these changes will be published?

**Comment by BigstickCarpet** [comment]
I just released v3.2.0, which includes the `browser` field in the `package.json` file

**Comment by iamakulov** [comment]
Thank you!
